### ISSUE TITLE
bigip execution/state modules should support folders/partitions

### ISSUE THREAD

**Issue by roskens** [saltstack/salt#40886]
### Description of Issue/Question
The bigip state module does not support folders other than Common. I should be able to do something like this for adding a new node into the 'internal' partition:

'add node /internal/node1':
  bigip.manage_node:
    - hostname: load_balancer
    - username: admin
    - password: admin
    - name: node1
    - address: 192.168.1.10
    - partition: internal

But I get a warning:

    Warnings: 'partition' is an invalid keyword argument for
              'bigip.manage_node'. If you were trying to pass additional data to
              be used in a template context, please populate 'context' with
              'key: value' pairs. Your approach will work until Salt Oxygen is
              out. Please update your state files.

Salt Version:
           Salt: 2016.11.4

Dependency Versions:
           cffi: Not Installed
       cherrypy: unknown
       dateutil: 2.5.3
      docker-py: Not Installed
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.8.1
        libgit2: 0.20.0
        libnacl: Not Installed
       M2Crypto: 0.20.2
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.2.3c1
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: 3.4.3
         pygit2: 0.20.3
         Python: 2.6.6 (r266:84292, Aug  9 2016, 06:11:56)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: 0.8.1
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: redhat 6.8 Santiago
        machine: x86_64
        release: 2.6.32-642.11.1.el6.x86_64
         system: Linux
        version: Red Hat Enterprise Linux Server 6.8 Santiago

**Comment by gtmanfred** [comment]
This would be great to have.  The last time I looked into this, there was a way to specify the partition in the node name

`name: ~partition~node`

But you would have to double check the f5 api on how to do that (it has been about 6 months since i looked at the f5 api docs.)

But we should totally have support for this.  If you wanted to submit a PR we would greatly appreciate it, otherwise I have tagged this as a feature request.

Thanks,
Daniel

**Comment by roskens** [comment]
I just tried that and it failed:

[ERROR   ] 01070088:3: The requested object name (/Common/~lb-qa~qa1) is invalid.
local:
----------
          ID: manage-node-lb-qa-qa1
    Function: bigip.manage_node
        Name: ~lb-qa~qa1
      Result: False
     Comment: 01070088:3: The requested object name (/Common/~lb-qa~qa1) is invalid.
     Started: 08:17:19.590366
    Duration: 153.376 ms
     Changes:

We are running 12.1.1 on our F5 LTMs.

**Comment by gtmanfred** [comment]
Yeah, i am not 100% sure that is how to specify the partition, but iirc i saw it in the api docs, so you might have to look that up to see if it will work.

As for this, I have marked it as a feature request.

Thanks!
Daniel

**Comment by gaetanquentin** [comment]
salt-call  bigip.list_pool hostname=LB_host username=xxx password=xxx name='~partition~poolname' works ok

**Comment by gaetanquentin** [comment]
we still need partition support in this module

**Comment by gaetanquentin** [comment]
no stale
